featuretools/tests/primitive_tests/test_feature_base.py

```diff
@@ -208,3 +210,32 @@ def test_multi_output_index_error(es):
     error_text = 'index is higher than the number of outputs'
     with pytest.raises(AssertionError, match=error_text):
         three_common[10]
+
+
+def test_rename(es):
+    feat = ft.Feature(es['log']['id'], parent_entity=es['sessions'], primitive=Count)
+    num_feature_names = len(feat.get_feature_names())
+    new_name = 'session_test'
+    new_names = [new_name] if num_feature_names == 1 else [new_name + '[{}]'.format(i) for i in range(num_feature_names)]
+    check_rename(feat, new_name, new_names)
+
+
+def test_rename_multioutput(es):
+    feat = ft.Feature(es['log']['product_id'],
+                      parent_entity=es['customers'],
+                      primitive=NMostCommon(n=2))
+    num_feature_names = len(feat.get_feature_names())
+    new_name = 'session_test'
+    new_names = [new_name] if num_feature_names == 1 else [new_name + '[{}]'.format(i) for i in range(num_feature_names)]
+    check_rename(feat, new_name, new_names)
+
+
+def test_rename_featureoutputslice(es):
+    multi_output_feat = ft.Feature(es['log']['product_id'],
+                                   parent_entity=es['customers'],
+                                   primitive=NMostCommon(n=2))
+    feat = ft.feature_base.FeatureOutputSlice(multi_output_feat, 0)
+    num_feature_names = len(feat.get_feature_names())
+    new_name = 'session_test'
+    new_names = [new_name] if num_feature_names == 1 else [new_name + '[{}]'.format(i) for i in range(num_feature_names)]
```

Comment:
since we know which features are multi output we can manually set `new_names` to the right value on a test by test basis in this file instead of relying on the output of `get_feature_names` to determine the correct value.